formula/src/main/java/com/instacart/formula/internal/FormulaManagerImpl.kt

```diff
@@ -39,12 +40,30 @@ internal class FormulaManagerImpl<Input, State, Output>(
 
     var transitionID: Long = 0
     var terminated = false
+    private var isEvaluating: Boolean = false
+
+    private val eventQueue = LinkedList<Event>()
+
+    fun onEvent(event: Event) {
+        if (terminated) {
+            event.dispatch()
+        } else if (isEvaluating) {
+            eventQueue.addLast(event)
+        } else {
+            delegate.onPostponedTransition(event)
+        }
+    }
+
+    fun hasPendingTransitions(): Boolean {
```

Comment:
Just to confirm that this is the critical piece of optimization in this PR.

Reply:
Yep, this is checked within `executeUpdates` block to know if we need to go back to `executeTransitions`